Adopt save and load related stuff of OrderList to trunk

diff -r 114ea429d960 src/saveload.cpp
--- a/src/saveload.cpp	Wed Oct 29 16:19:16 2008 +0100
+++ b/src/saveload.cpp	Wed Oct 29 16:19:21 2008 +0100
@@ -37,7 +37,7 @@
 
 #include "table/strings.h"
 
-extern const uint16 SAVEGAME_VERSION = 102;
+extern const uint16 SAVEGAME_VERSION = 103;
 
 SavegameType _savegame_type; ///< type of savegame we are loading
 
diff -r 114ea429d960 src/saveload.h
--- a/src/saveload.h	Wed Oct 29 16:19:16 2008 +0100
+++ b/src/saveload.h	Wed Oct 29 16:19:21 2008 +0100
@@ -69,7 +69,6 @@ enum SLRefType {
 };
 
 #define SL_MAX_VERSION 255
-#define SL_OL_VER (SL_MAX_VERSION)
 
 enum {
 	INC_VEHICLE_COMMON = 0,
diff -r 114ea429d960 src/vehicle.cpp
--- a/src/vehicle.cpp	Wed Oct 29 16:19:16 2008 +0100
+++ b/src/vehicle.cpp	Wed Oct 29 16:19:21 2008 +0100
@@ -251,7 +251,7 @@ void AfterLoadVehicles(bool part_of_load
 
 		FOR_ALL_VEHICLES(v) {
 			if (v->orders.old != NULL) {
-				if (CheckSavegameVersion(SL_OL_VER)) { // Pre-104 didn't save an OrderList
+				if (CheckSavegameVersion(103)) { // Pre-103 didn't save an OrderList
 					if (mapping[v->orders.old] == NULL) {
 						/* This adds the whole shared vehicle chain for case b */
 						v->orders.list = mapping[v->orders.old] = new OrderList(v->orders.old, v);
@@ -2216,7 +2216,7 @@ static const SaveLoad _common_veh_desc[]
 
 	    SLE_VAR(Vehicle, cur_order_index,      SLE_UINT8),
 	/* num_orders is now part of OrderList and is not saved but counted */
-	SLE_CONDNULL(1,                                                          0, SL_OL_VER-1),
+	SLE_CONDNULL(1,                                                          0, 102),
 
 	/* This next line is for version 4 and prior compatibility.. it temporarily reads
 	    type and flags (which were both 4 bits) into type. Later on this is
@@ -2237,8 +2237,8 @@ static const SaveLoad _common_veh_desc[]
 	SLE_CONDVARX(cpp_offsetof(Vehicle, current_order) + cpp_offsetof(Order, wait_time),      SLE_UINT16, 67, SL_MAX_VERSION),
 	SLE_CONDVARX(cpp_offsetof(Vehicle, current_order) + cpp_offsetof(Order, travel_time),    SLE_UINT16, 67, SL_MAX_VERSION),
 
-	SLE_CONDREF(Vehicle, orders,               REF_ORDER,                   0, SL_OL_VER-1),
-	SLE_CONDREF(Vehicle, orders,               REF_ORDERLIST,              SL_OL_VER, SL_MAX_VERSION),
+	SLE_CONDREF(Vehicle, orders,               REF_ORDER,                   0, 102),
+	SLE_CONDREF(Vehicle, orders,               REF_ORDERLIST,             103, SL_MAX_VERSION),
 
 	SLE_CONDVAR(Vehicle, age,                  SLE_FILE_U16 | SLE_VAR_I32,  0, 30),
 	SLE_CONDVAR(Vehicle, age,                  SLE_INT32,                  31, SL_MAX_VERSION),
